src/server/replica.cc

```diff
@@ -585,6 +606,10 @@ error_code Replica::ConsumeDflyStream() {
 
   JoinAllFlows();
 
+  // TODO: Vlad I want to make sure we dont need cleanup here of shared map and flow queues
+  // If I understand the reason we dont have cleanup is because after we exit here replica class
+  // is distroyed?
+
```

Comment:
What kind of cleanup? The current cleanup for a flow done by the stable sync function is just joining it

Reply:
Because the replica is holding state than I would think that after each step we will have a cleanup for the step, f.e in here clear the shared transaction map and queues in each flow. The reason for it is if the step failed to  and you want to do recover I would want to have a clean data in replica class. 
Unless we are destroying the class after the step fails, as currently done in full sync if we start it again you recreate the replica flow classes.
So what is the flow if the replica connection fails while we are in stable state? the error handler is invoked right? and after that if we are able to connect again today we will start the full sync step again right? this will recreate the replica flow classes. But what if we want to continue from where we failed if possible ?
Maybe I should not think about this now and we will get to it once we need to do the partial sync

Reply:
Yes, I would suggest thinking about this when we do partial sync. We can just trivially drop all the data we have inside the transaction map, etc...